support/build/Makefile.rules

```diff
@@ -830,28 +850,26 @@ endef
 #
 # preprule_libobj $1:libname,$2:source,$3:preptargetsuffix,$4:variant(optional)
 define preprule_libobj =
-$(eval $(call preprule,$(1),$(2),$(3),\
-  $($(call vprefix_src,$(1),$(2),PREPINCLUDES))\
-  $($(call vprefix_src,$(1),$(2),PREPINCLUDES-y))\
-  $($(call vprefix_src,$(1),$(2),PREPFLAGS))\
-  $($(call vprefix_src,$(1),$(2),PREPFLAGS-y))\
-  $(if $(4),\
-   $($(call vprefix_srcv,$(1),$(2),$(4),PREPINCLUDES))\
-   $($(call vprefix_srcv,$(1),$(2),$(4),PREPINCLUDES-y))\
-   $($(call vprefix_srcv,$(1),$(2),$(4),PREPFLAGS))\
-   $($(call vprefix_srcv,$(1),$(2),$(4),PREPFLAGS-y))\
-  ,)\
+$(eval subbuild=$($(call vprefix_src,$(1),$(2),$(4),SUBBUILD)))
```

Comment:
I went through the same thoughts and chose this name because it was there already (also my mistake) and I was able to keep the variable suffix short. I wanted to avoid anything that has an underscore because resolving uses a lot of underscores already: 
`<LIBNAME>_<SOURCENAME>_<VARIANTNAME>_SUBBUILD`
The thing is that the variable suffix (`_SUBBUILD`) will be used in `Makefile.uk`'s in the future while it is unlikely that somebody uses `mk_sub_build_dir` directly (although there might be exceptions). We can still choose a better name. I also had (something with) `TARGET` in mind... :thinking: